src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -189,6 +191,20 @@ public override void OnError(Exception ex)
                 TraceCompilationDiagnostics(exc.Diagnostics, LogTargets.User);
                 throw;
             }
+            catch (CompilerErrorException exc)
+            {
+                TraceOnPrimaryHost("Script compiler error", TraceLevel.Error, _compilationService.GetType().Name, exc);
```

Comment:
On first glance, this sounds like I've got an issue with my code compiling -- can we make it clear that that's not the case?